src/app/public/src/modules/shared/route.service.ts

```diff
@@ -126,36 +126,55 @@ export class StacheRouteService {
   }
 
   private sortRoutes(routes: StacheNavLink[]): StacheNavLink[] {
-
-    const sortByName = (a: any, b: any): number  => {
-      if (a.name.toLowerCase() < b.name.toLowerCase()) {
-        return -1;
-      } else if (a.name.toLowerCase() > b.name.toLowerCase()) {
-        return 1;
-      } else {
-        return 0;
-      }
-    };
-
-    const sortByOrder = (a: any, b: any): number => {
-      if (a.order < b.order) {
-        return -1;
-      } else if (a.order > b.order) {
-        return 1;
+    const sortedRoutes = routes.map(this.validateNavOrder)
```

Comment:
The route order is derived from the `StacheRouteMetadataService`, correct? I feel like the `order` should always be "valid" if it is fetched from the service. Maybe include the validation methods here?
https://github.com/blackbaud/stache2/blob/master/src/app/public/src/modules/shared/route-metadata.service.ts#L23